### PR TITLE
 chore(i18n): add e2e tests for i18n

### PR DESCRIPTION
**Pull request by Sayan751**
This PR adds the e2e tests for i18n. The tests includes the following.

- translations via html attributes, value converter, and binding behavior
- translations via code
- change of translations on change of locale

Additionally, it removes the deprecated babel/polyfill, as the package is deprected and no longer available for download with the latest dependency graph. On the other hand, the deprectaed dependency caused CI pipeline to fail. Removing this should potentially fix that problem.

### CODE REVIEW

**Comment by zewa666** [comment]
This is fantastic. Im on the phone so cant go into a detailed look but from a quick peek you could group related tests, which you added a region at, with a nested describe. This would also help to quickly only run said region with fdescribe/describe.only

**Comment by zewa666** [comment]
Looking at the last test its basically a repetition of all previous ones after the locale change.

We could simplify that by using databound tests. Since every it is merely a function you can loop over an array of locales and then inside have the its placed. The results then are stored together with the selector and expected values per languages.

**Comment by fkleuver** [comment]
Great start, I like it a lot.

As @zewa666 already pointed out, there is some repetition which makes a case for generating some tests in some form or shape.

There's really two ways to do this, we're employing both ways in vNext depending on the needs of the situation.

The easiest to implement (but harder to debug and understand the specs) is this for example:
https://github.com/aurelia/aurelia/blob/master/packages/__tests__/jit/expression-parser.spec.ts#L431

The tests are generated at runtime with simple loops. This is in fact how most of vNext is tested.

The other approach, generating the files upfront, is harder to implement, but keeps the tests transparent and easy to debug.
We're doing this with the more complex integration tests to keep them easy to debug.

The generator code, arguably a bit convoluted for this purpose: https://github.com/aurelia/aurelia/blob/master/scripts/generate-tests/template-compiler.static.ts#L868
Some of the generated tests:
https://github.com/aurelia/aurelia/blob/master/packages/__tests__/jit-html/generated/template-compiler.static.if-else.repeat.spec.ts

As far as I'm concerned it's completely up to you whether or not you'll employ any of these methods. From a testing approach perspective this PR looks fine to me, and @zewa666 still needs to take a look regarding the covered API surface I think?

Just wanted to point it out / make you aware of the concept.

**Comment by Sayan751** [comment]
I like the idea of generating the tests using loop and an array of expected input and output. I use that concept on a daily basis actually. In this case, i thought that it is better to keep every tests separate. I will take a look if the code can be refactored to be more compact and still keep it simple. 
On the other hand, generating the files seems bit complex, and I would like to avoid that for this purpose.

**Comment by zewa666** [comment]
Can we turn off video recording? no real value, since we can run in debug anyway. Either by an CLI [override](https://docs.cypress.io/guides/references/configuration.html#Overriding-Options) or setting `video: false` in the cypress.json config file.

File: src/resources/i18n-target.html [hunk 1]
```diff
@@ -0,0 +1,52 @@
+<template>
```
Comment (zewa666) [inline src/resources/i18n-target.html]: I'd move this file into `resources/plugins/i18n/i18n-target.html`. This way we can start a proper separation if the same app is going to be used for various other plugins.

File: src/resources/i18n-target.html [hunk 2]
```diff
@@ -0,0 +1,52 @@
+<template>
+
+  <button id="locale-changer-de" click.delegate="changeLocale('de')">DE</button>
+
+  <span id="i18n-simple" t="simple.text;[title]simple.attr"></span>
```
Comment (zewa666) [inline src/resources/i18n-target.html]: simple is actually performing two tests on multiple features:

* simple interpolation
* attribute overrides
* multiple targets with semicolon

File: src/resources/i18n-target.html [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+  <button id="locale-changer-de" click.delegate="changeLocale('de')">DE</button>
+
+  <span id="i18n-simple" t="simple.text;[title]simple.attr"></span>
+
+  <span id="i18n-ctx" t="status"></span>
+  <span id="i18n-ctx-dispatched" t="status" t-params.bind="{context: 'dispatched', date: dispatchedOn}"></span>
+  <span id="i18n-ctx-delivered" t="status" t-params.bind="{context: 'delivered', date: deliveredOn}"></span>
+
+  <span id="i18n-items-0" t="itemWithCount" t-params.bind="{count: 0}"></span>
```
Comment (zewa666) [inline src/resources/i18n-target.html]: I liked the idea that above the `id` is pretty much telling which feature we're testing. Maybe here we could include `count` or `plural` into the id.

File: src/resources/i18n-target.html [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  <span id="i18n-nf-vc"> ${ 123456789.12 | nf : undefined : locale} </span>
+  <span id="i18n-nf-bb"> ${ 123456789.12 & nf : undefined : 'de'} </span>
+
+  <span id="i18n-nf-vc-cur"> ${ 123456789.12 | nf: {style:'currency', currency: 'EUR' } : 'de' } </span>
+  <span id="i18n-nf-bb-cur"> ${ 123456789.12 & nf: {style:'currency', currency: 'USD' } : locale } </span>
+
+  <span id="i18n-df-vc"> ${ dispatchedOn | df : undefined : locale} </span>
+  <span id="i18n-df-bb"> ${ dispatchedOn & df : undefined : 'de'} </span>
+
+  <span id="i18n-rt-vc"> ${ myDate | rt} </span>
```
Comment (zewa666) [inline src/resources/i18n-target.html]: one more test for the binding behavior custom signal `aurelia-relativetime-signal` could be nice here
https://github.com/aurelia/i18n/blob/master/test/unit/relative.time.spec.ts#L222

File: src/resources/i18n-target.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import { I18N, RelativeTime } from "aurelia-i18n";
+
+@customElement("i18n-target")
+@autoinject
+export class I18NTarget {
+    dispatchedOn = new Date(2020, 1, 10, 5, 15);
+    deliveredOn = new Date(2021, 1, 10, 5, 15);
+    myDate: Date;
+    locale: string;
+    translations: any;
```
Comment (zewa666) [inline src/resources/i18n-target.ts]: `{ [key: string]: string }` instead of any

File: src/resources/i18n-target.html [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  <span id="i18n-prepend-append" t="[prepend]pretest;[append]post-test">Blue</span>
+
+  <span id="i18n-t-vc"> ${'itemWithCount' | t : {count: 10}} </span>
+  <span id="i18n-t-bb"> ${'itemWithCount' & t : {count: 100}} </span>
+
+  <span id="i18n-nf-vc"> ${ 123456789.12 | nf : undefined : locale} </span>
+  <span id="i18n-nf-bb"> ${ 123456789.12 & nf : undefined : 'de'} </span>
+
+  <span id="i18n-nf-vc-cur"> ${ 123456789.12 | nf: {style:'currency', currency: 'EUR' } : 'de' } </span>
+  <span id="i18n-nf-bb-cur"> ${ 123456789.12 & nf: {style:'currency', currency: 'USD' } : locale } </span>
```
Comment (zewa666) [inline src/resources/i18n-target.html]: at least one of the unformat features could be tested here as well
https://github.com/aurelia/i18n/blob/master/test/unit/numberformat.spec.ts#L43

File: src/resources/i18n-target.html [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  <span id="i18n-interval-1" t="itemWithCount_interval" t-params.bind="{postProcess: 'interval', count: 1}"></span>
+  <span id="i18n-interval-2" t="itemWithCount_interval" t-params.bind="{postProcess: 'interval', count: 2}"></span>
+  <span id="i18n-interval-3" t="itemWithCount_interval" t-params.bind="{postProcess: 'interval', count: 3}"></span>
+  <span id="i18n-interval-6" t="itemWithCount_interval" t-params.bind="{postProcess: 'interval', count: 6}"></span>
+  <span id="i18n-interval-7" t="itemWithCount_interval" t-params.bind="{postProcess: 'interval', count: 7}"></span>
+  <span id="i18n-interval-10" t="itemWithCount_interval" t-params.bind="{postProcess: 'interval', count: 10}"></span>
+
+  <span id="i18n-html" t="[html]html"></span>
+
+  <span id="i18n-prepend-append" t="[prepend]pretest;[append]post-test">Blue</span>
```
Comment (zewa666) [inline src/resources/i18n-target.html]: multiple attributes with same key
`[placeholder,aria-placeholder]placeholder`

File: src/resources/i18n-target.html [hunk 8]
```diff
@@ -0,0 +1,52 @@
+<template>
+
+  <button id="locale-changer-de" click.delegate="changeLocale('de')">DE</button>
+
+  <span id="i18n-simple" t="simple.text;[title]simple.attr"></span>
+
+  <span id="i18n-ctx" t="status"></span>
```
Comment (zewa666) [inline src/resources/i18n-target.html]: nested translations
`<span t="$t(title) $t(subtitle)">Title subtitle</span>`

File: src/resources/i18n-target.html [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+  <span id="i18n-interval-0" t="itemWithCount_interval" t-params.bind="{postProcess: 'interval', count: 0}"></span>
+  <span id="i18n-interval-1" t="itemWithCount_interval" t-params.bind="{postProcess: 'interval', count: 1}"></span>
+  <span id="i18n-interval-2" t="itemWithCount_interval" t-params.bind="{postProcess: 'interval', count: 2}"></span>
+  <span id="i18n-interval-3" t="itemWithCount_interval" t-params.bind="{postProcess: 'interval', count: 3}"></span>
+  <span id="i18n-interval-6" t="itemWithCount_interval" t-params.bind="{postProcess: 'interval', count: 6}"></span>
+  <span id="i18n-interval-7" t="itemWithCount_interval" t-params.bind="{postProcess: 'interval', count: 7}"></span>
+  <span id="i18n-interval-10" t="itemWithCount_interval" t-params.bind="{postProcess: 'interval', count: 10}"></span>
+
+  <span id="i18n-html" t="[html]html"></span>
```
Comment (zewa666) [inline src/resources/i18n-target.html]: two tests for images could be usesful

1. default translation targets the src attribute
2. data-src to be used as original url

(See docs "Translating Images" https://aurelia.io/docs/plugins/i18n#using-the-plugin)

File: src/resources/i18n-target.html [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+  <!-- translations via code -->
+  <div>
+    <span id="i18n-code-simple">${translations.simple}</span>
+    <span id="i18n-code-context">${translations.context}</span>
+    <span id="i18n-code-plural">${translations.plural}</span>
+    <span id="i18n-code-interval">${translations.interval}</span>
+    <span id="i18n-code-num">${translations.num}</span>
+    <span id="i18n-code-date">${translations.date}</span>
+    <span id="i18n-code-rtime">${translations.rtime}</span>
```
Comment (zewa666) [inline src/resources/i18n-target.html]: `globalVars` placeholder feature could be tested as well
https://github.com/aurelia/i18n/blob/master/test/unit/feature.placeholders.spec.ts

File: src/resources/i18n-target.html (hunk 2 above)
Comment (Sayan751) [reply src/resources/i18n-target.html]: Done

File: src/resources/i18n-target.html (hunk 3 above)
Comment (Sayan751) [reply src/resources/i18n-target.html]: Done

File: src/resources/i18n-target.html (hunk 1 above)
Comment (Sayan751) [reply src/resources/i18n-target.html]: Done

File: src/resources/i18n-target.html (hunk 4 above)
Comment (Sayan751) [reply src/resources/i18n-target.html]: Done

File: src/resources/i18n-target.ts (hunk 5 above)
Comment (Sayan751) [reply src/resources/i18n-target.ts]: Done

File: src/resources/i18n-target.html (hunk 6 above)
Comment (Sayan751) [reply src/resources/i18n-target.html]: Done

File: src/resources/i18n-target.html (hunk 7 above)
Comment (Sayan751) [reply src/resources/i18n-target.html]: Done

File: src/resources/i18n-target.html (hunk 8 above)
Comment (Sayan751) [reply src/resources/i18n-target.html]: Done

File: src/resources/i18n-target.html (hunk 9 above)
Comment (Sayan751) [reply src/resources/i18n-target.html]: Done for the default `src` translation. Skipped the `data-src` part as it does not get any special treatment.

File: src/resources/i18n-target.html (hunk 10 above)
Comment (Sayan751) [reply src/resources/i18n-target.html]: Done

**Comment by Sayan751** [comment]
@zewa666 Requesting a review

**Comment by zewa666** [comment]
LGTM

**Comment by zewa666** [comment]
@fkleuver do you want to wire up these tests with a CI provider to have future PRs checked with them or should we stay with manual execution?

**Comment by Sayan751** [comment]
As far as I know this repo is already used in CI pipeline.

**Comment by zewa666** [comment]
Ready for merge @eisenbergeffect

**Pull request merged**